jraft-core/src/main/java/com/alipay/sofa/jraft/util/ThreadPoolGroup.java

```diff
@@ -0,0 +1,115 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.jraft.util;
+
+import com.alipay.sofa.jraft.Closure;
+import com.alipay.sofa.jraft.Status;
+import com.codahale.metrics.MetricRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.*;
+
+/**
+ * ThreadPool based on Raft-Group isolation
+ *
+ * @author tynan.liu
+ * @date 2022/6/24 17:25
+ **/
+public class ThreadPoolGroup {
+    private static final Logger                                    LOG                      = LoggerFactory
+                                                                                                .getLogger(ThreadPoolGroup.class);
+    private static final ConcurrentMap<String, ThreadPoolExecutor> GROUP_THREAD_POOL_ROUTER = new ConcurrentHashMap<>();
+    private static volatile ThreadPoolExecutor                     DEFAULT_GLOBAL_THREAD_POOL;
+
+    public static void registerThreadPool(final MetricRegistry registry, String groupId, ThreadPoolExecutor executor) {
+        if (GROUP_THREAD_POOL_ROUTER.containsKey(groupId)) {
+            LOG.warn("The group:{} has already registered the ThreadPool", groupId);
+            return;
+        }
+
+        synchronized (GROUP_THREAD_POOL_ROUTER) {
+            if (GROUP_THREAD_POOL_ROUTER.containsKey(groupId)) {
+                LOG.warn("The group:{} concurrently registers the ThreadPool", groupId);
+                return;
+            }
+            if (executor == null) {
+                if (DEFAULT_GLOBAL_THREAD_POOL == null) {
+                    DEFAULT_GLOBAL_THREAD_POOL = ThreadPoolUtil.newBuilder().poolName("JRAFT_GROUP_DEFAULT_EXECUTOR")
+                        .enableMetric(true).coreThreads(Utils.MIN_CLOSURE_EXECUTOR_POOL_SIZE)
+                        .maximumThreads(Utils.MAX_CLOSURE_EXECUTOR_POOL_SIZE).keepAliveSeconds(60L)
+                        .workQueue(new SynchronousQueue<>())
+                        .threadFactory(new NamedThreadFactory("JRaft-Group-Default-Executor-", true)).build();
+                }
+                executor = DEFAULT_GLOBAL_THREAD_POOL;
+            }
+            GROUP_THREAD_POOL_ROUTER.putIfAbsent(groupId, executor);
+            if (registry != null) {
+                registerClosureExecutorMetrics(registry, groupId, executor);
+            }
+        }
+    }
+
+    private static void registerClosureExecutorMetrics(final MetricRegistry registry, String groupId,
```

Comment:
It's the API user's responsibility to register metric set, we don't need to implement it in this class.